zignal/audio.py

```diff
@@ -537,24 +537,26 @@ def write_wav_file(self, filename=None):
         except:
             self._logger.exception("Could not write file: '%s'" %filename)
 
-    def plot(self, ch=1, plotname=None, **kwargs):
+    def plot(self, ch=1, plotname=None, plotrange=None, **kwargs):
         """Plot the audio data on a time domain plot.
 
         example:
 
             x1 = Sinetone(f0=0.2, fs=10, nofsamples=50)
-            x1.plot(linestyle='--', marker='x', color='r', label='sine at 0.2Hz')
+            x1.plot(plotrange=range(0,10), linestyle='--', marker='x', color='r', label='sine at 0.2Hz')
```

Comment:
I don't like that you have to use the range object, it isn't intuitive and not in line with the other methods in this class. I rather have a tuple with a start and stop time instead, so that the argument would be `plotrange=(0, 10)` instead. Also, I think that the number specified should be in seconds, not in samples. The `self.plot(...)` method is a convenience method and you shouldn't have to care about the sample rate when selecting to plot something. If you really care about the exact plot in samples then this convenience method is probably not good enough anyway and it is always possible to plot the raw data outside of this class. But internally in this method I think it is great to use a `range(...)` or `slice(...)` to select the section of the array to plot.

I would like to use the plot method like in the example code below:

```
import zignal

if __name__ == '__main__':    
    fs = 16000
    a = zignal.Audio(channels=0, fs=fs, duration=6)
    b = zignal.Sinetones(1, 4, 12, fs=fs, duration=2)
    a.append(b)
    print(a)

    a.delay(fs*2, channel=2)
    a.delay(fs*4, channel=3)
    
    c = a.to_mono()
    print(c)
    
    c.plot() # like today, first channel, from start to stop
    
    c.plot(plotrange=(1.6, 5.1))    # between time in seconds
    c.plot(plotrange=(None, 5.1))   # from the beginning to some time in seconds
    c.plot(plotrange=(4, None))     # from some time in seconds to the end
    
    a.plot(ch="all", plotrange=(1.5, 3))    # all channels, between range in seconds
    a.plot(ch="all", plotrange=(None, 3))   # from the beginning 
    a.plot(ch="all", plotrange=(1.5, None)) # to the end    
```

I realise that the `delay` method actually takes samples as argument and not time, so the methods aren't consistent :) But still, I think it should be seconds (as float). The default value could then be `plotrange=(None, None)` which also would hint how to use this.

Reply:
It shouldn't be difficult for me to adjust `plotrange` to take seconds instead of samples as long as the sample rate is known.

Regarding `delay`, I think it's appropriate to leave it taking samples as the argument because it needs to append that many samples anyway, and specifying it as seconds could make it imprecise, as in the code would have to estimate how many samples it would need to insert, since the user could specify an arbitrary time argument in seconds that doesn't exactly match a range of samples.

By the way, when will the bugfixes in the previous PR be pushed to PyPI?